capa/features/extractors/binexport2/arch/intel/insn.py

```diff
@@ -91,66 +128,36 @@ def extract_insn_offset_features(
     ii: InstructionContext = ih.inner
 
     be2: BinExport2 = fhi.ctx.be2
-    instruction: BinExport2.Instruction = be2.instruction[ii.instruction_index]
-
-    if len(instruction.operand_index) == 0:
-        # skip things like:
-        #   .text:0040116e leave
-        return
-
-    mnemonic: str = get_instruction_mnemonic(be2, instruction)
-    value: int
 
-    for i, operand_index in enumerate(instruction.operand_index):
-        operand: BinExport2.Operand = be2.operand[operand_index]
-
-        is_dereference = False
-        for expression_index in operand.expression_index:
-            if be2.expression[expression_index].type == BinExport2.Expression.DEREFERENCE:
-                is_dereference = True
-                break
-
-        if not is_dereference:
-            continue
-
-        phrase_info: Optional[OperandPhraseInfo] = get_operand_phrase_info(be2, operand)
-        if not phrase_info:
-            continue
-
-        if phrase_info.displacement:
-            if phrase_info.base and phrase_info.base.symbol.lower().endswith(("bp", "sp")):
-                # skips things like:
-                # 00401068 MOV dword ptr [EBP + local_8],EAX
-                continue
-
-            value = mask_immediate(fhi.arch, phrase_info.displacement.immediate)
-            if not is_address_mapped(be2, value):
-                value = capa.features.extractors.binexport2.helpers.twos_complement(fhi.arch, value, 32)
+    match = OFFSET_PATTERNS.match_with_be2(be2, ii.instruction_index)
+    if not match:
+        match = OFFSET_ZERO_PATTERNS.match_with_be2(be2, ii.instruction_index)
+        if not match:
+            return
 
-                yield Offset(value), ih.address
-                yield OperandOffset(i, value), ih.address
+        yield Offset(0), ih.address
+        yield OperandOffset(match.operand_index, 0), ih.address
```

Comment:
looks good, but general question: do we care about 0 offset features?

Reply:
probably not? i saw that the other extractors handled this case, so i duplicated it here too.

Reply:
right, let's leave it for consistency then